packages/pkg/src/index.ts

```diff
@@ -39,10 +43,25 @@ export default {
         const { filepath } = options
         const ast = parse(text, options) as { node: ObjectExpression }
         if (PKG_REG.test(filepath)) {
-          ast.node.properties = format(ast.node.properties)
+          ast.node.properties = format(
+            ast.node.properties,
+            options as FormatOptions,
+          )
         }
         return ast
       },
     },
   },
+  options: {
+    packageSortOrder: {
+      since: '0.21.0',
+      category: 'Package',
+      type: 'string',
+      array: true,
+      default: [{ value: [] }],
```

Comment:
Why this is needed?

Reply:
It seems wierd to me. cc @fisker

Reply:
I don't know! What did the error message say? Received an `undefined`?

Reply:
Even use `default: [{ value: [] }],`, we are not using an copy, it can be changed during parse/print. 🤣

Reply:
> This part is a mess, and hard to change

Indeed, if it fails, it fails. 😂